tribe-ext-community-currency-symbol/src/Tribe/Settings_Helper.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+// Do not load directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+if ( class_exists( 'Tribe__Extension__Settings_Helper' ) ) {
+	return;
+}
+
+/**
+ * Helper for inserting/removing fields on the WP Admin Tribe Settings pages
+ */
+class Tribe__Extension__Settings_Helper {
+
+	/**
+	 * Fields inserted into misc section
+	 *
+	 * @var array
+	 */
+	private $insert_fields_misc = array();
+
+	/**
+	 * Fields that will be inserted above a specified field
+	 *
+	 * @var array
+	 */
+	private $insert_fields_above = array();
+
+	/**
+	 * Fields that will be inserted below a specified field
+	 *
+	 * @var array
+	 */
+	private $insert_fields_below = array();
+
+	/**
+	 * Array of settings being added to a Tribe Settings tab
+	 *
+	 * @var array
+	 */
+	private $remove_fields = array();
+
+
+	/**
+	 * Setup the helper
+	 *
+	 * @param int $priority Priority at which this hooks into 'tribe_settings_tab_fields'.
+	 */
+	public function __construct( $priority = 100 ) {
+		add_filter( 'tribe_settings_tab_fields', array( $this, 'filter_options' ), $priority, 2 );
+	}
+
+
+	/**
+	 * Add a field to a Tribe Settings tab
+	 *
+	 * @param string $field_key         Option key for your setting. Example: 'fancyOptionName'.
+	 * @param array $field_args         See Tribe__Field() for available args.
+	 *                                  Example: array( 'type' => 'checkbox_bool, 'label' => ... )
+	 * @param string $setting_tab       Settings tab where this will be added. Example: 'display'.
+	 * @param string $neighboring_field (optional) The field key/HTML name="" attribute to insert this under.
+	 * @param bool   $above             (optional) Insert above or below its neighbor.
+	 */
+	public function add_field( $field_key, $field_args, $setting_tab, $neighboring_field = null, $above = true ) {
+		// Our settings walker needs 'key' => arg pairs.
+		$field = array( $field_key => $field_args );
+
+		$this->add_fields( $field, $setting_tab, $neighboring_field, $above );
+	}
+
+	/**
+	 * Add multiple fields to a Tribe Settings tab
+	 *
+	 * @param array  $fields            Fields that will be added, expects 'fieldname' => (array) args.
+	 * @param string $setting_tab       Settings tab where this will be added. Example: 'display'.
+	 * @param string $neighboring_field (optional) The field key/HTML name="" attribute to insert this under.
+	 * @param bool   $above             (optional) Insert above or below its neighbor.
+	 */
+	public function add_fields( $fields, $setting_tab, $neighboring_field = null, $above = false ) {
+
+		foreach ( $fields as $key => $val ) {
+			// Handles button_link type that doesn't exist in Tribe__Field.
+			if ( 'button_link' === $fields[ $key ]['type'] ) {

```

Comment:
When inside of a Loop, it makes it more readable when you invert the conditional to call a continue, it basically avoids an indentation level.

Also using `&$field` on the `foreach` allows the code to be less verbose without loosing readability, due to making `$field` be an alias to `$fields[ $key ]`.

```php
foreach ( $fields as $key => &$field ) {
	// Handles button_link type that doesn't exist in Tribe__Field.
	if ( 'button_link' !== $field['type'] ) {
		continue;
	}

	$field['type'] = 'html';
	$field['html'] = $this->get_button_html(
		$key,
		$field['label'],
		$field['url'],
		$field['tooltip']
	);
	unset( $field['label'] );
	unset( $field['tooltip'] );
}
```

Reply:
@bordoni, I passed this feedback onto @elimn because this helper class is making its way into common via maintenance release. So no change from me here, since this helper class is soon to be the legacy way to do this.